### PR TITLE
TypeScript

### PR DESCRIPTION
**Pull request by zautumnz**
## Change Type

* [x] Feature
* [ ] Chore
* [ ] Bug Fix

## Change Level

* [ ] major
* [x] minor
* [ ] patch

## Checklist

* [ ] Followed PR template
* [ ] Added unit tests if applicable
* [ ] Added contract tests if applicable
* [ ] Added integration tests if applicable
* [ ] Added e2e tests if applicable

### CODE REVIEW

**Comment by zacanger** [comment]
The tsconfig was auto generated by react-scripts, I guess. All I did was install typescript and some types packages, and the next time I ran `build` it was there.

**Comment by TomPridham** [comment]
i might copy that config. it seems like its got some good defaults

File: src/app.tsx [hunk 1]
```diff
@@ -13,18 +13,40 @@ html, body {
...
+  name: string,
+  stargazers_count: number,
+}
+
+type State = {
+  repos: Array<DisplayRepo>
+}
+
+export default class App extends React.Component<{}, State> {
+  constructor(props: any) {
```
Comment (TomPridham) [inline src/app.tsx]: is this not a lint error?

File: src/app.tsx [hunk 2]
```diff
@@ -13,18 +13,40 @@ html, body {
...
   }
 
   getFromStorage() {
     try {
-      const data = JSON.parse(localStorage.getItem(storageKey))
+      const data = JSON.parse(localStorage.getItem(storageKey) || '""')
       const now = new Date()
       const then = new Date(data.date)
-      if ((now - then) / 36e5 <= 24) {
+      if ((+now - +then) / 36e5 <= 24) {
```
Comment (TomPridham) [inline src/app.tsx]: i get what this is doing, but i don't really like it

File: src/app.tsx (hunk 1 above)
Comment (zacanger) [reply src/app.tsx]: This isn't using our lint config, it's using the built in CRA one I guess

File: src/app.tsx (hunk 2 above)
Comment (zacanger) [reply src/app.tsx]: I don't love it, but TS doesn't let you do math with plain old dates

**Review by TomPridham — APPROVED** [review]

**Pull request merged**